### PR TITLE
#342 | PomChangePlugin feature

### PR DESCRIPTION
**Pull request by praveenksingh**
This adds the functionality to add/update/remove the following for a plugin from pom files.
- version
- extensions
- executions
- dependencies (plugin dependencies)

#342

### CODE REVIEW

File: butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+     * Operation to set the Executions that will add/update the existing plugin executions.
+     * @param   executions  List of {@link PluginExecution} that needs to be set in the transformed plugin
+     * @return              this PomChangePlugin object
+     */
+    public PomChangePlugin setExecutions(List<PluginExecution> executions) {
+        checkForNull("Executions", executions);
+        Map<String, PluginExecution> executionMap = new LinkedHashMap<>();
+        for (PluginExecution exec : executions) {
+            if (executionMap.containsKey(exec.getId())) {
+                throw new IllegalStateException("You cannot have two plugin executions with the same " +
```
Comment (fabiocarvalho777) [inline butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java]: Make this an `IllegalArgumentException` please

File: butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Changes a Plugin in a Maven POM file.
+ * It allows changing version, extensions, executions and dependencies but not group id and artifact id.
+ * It also allows removing specific configuration, letting them
+ * to have default values, or be managed by applicable.
+ * <br>
+ * If the plugin to be changed doesn't actually exist, it will result
```
Comment (fabiocarvalho777) [inline butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java]: This should depend actually on the user. That is why this class implements `ChangeOrRemoveElement`. See `IfNotPresent` instance variable. Please change this comment and honor the `ChangeOrRemoveElement` interface.

File: butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java (hunk 1 above)
Comment (praveenksingh) [reply butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java]: Makes sense. Fixed this pushed code with `IllegalArgumentException`

File: butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java (hunk 2 above)
Comment (praveenksingh) [reply butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java]: Updated the comment, where it mentions that the default behavior can be changed using `IfNotPresent`

**Pull request merged**